doc/node_config_json_schema.json

```diff
@@ -0,0 +1,609 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "debug-brk": {
```

Comment:
Can you add a sort-keys eslint rule for this file?

Reply:
I don't think we actually need this file as much as we need a way of generating it either during build or using a command-line flag similar to what we have for the bash completion script. This information can be sourced directly from the node_options metadata without having to maintain this separate file.

Reply:
At least for reviewing this PR, it's good to have the file committed so we can comment on it.

Reply:
the file is now tested against and can be auto generated